deps/graph-parser/src/logseq/graph_parser/block.cljs

```diff
@@ -592,6 +592,28 @@
                 (assoc :block/updated-at updated-at))]
     (dissoc block :title :body :anchor)))
 
+(defn fix-duplicate-id
+  [block]
+  (-> block
+      (assoc :uuid (d/squuid))
+      (update :properties dissoc :id)
+      (update :properties-text-values dissoc :id)
+      (update :properties-order #(vec (remove #{:id} %)))
+      (update :content (fn [c]
+                         (let [replace-str (re-pattern
+                                            (str
+                                             "\n*\\s*"
+                                             (if (= :markdown (:format block))
+                                               (str "id:: " (:uuid block))
+                                               (str ":id: " (:uuid block)))))]
+                           (string/replace c replace-str ""))))))
+
+(defn block-exists-in-another-page?
+  [db block-uuid current-page-name]
+  (when (and db current-page-name)
```

Comment:
Currently the :page-name option is only passed via `extract/extract-pages-and-blocks`. Looks like all other callers will ignore this fix. I don't know if that's desired behavior but something to be aware of

Reply:
Good point! [d109045](https://github.com/logseq/logseq/pull/8668/commits/d109045cf867899228ad03165ea3eee6cc7bb39c) handles the other cases.

Reply:
Great!